OpenCVExample/OpenCVExample/MainPage.xaml.cpp

```diff
@@ -0,0 +1,144 @@
+﻿//
+// MainPage.xaml.cpp
+// Implementation of the MainPage class.
+//
+
+#include "pch.h"
+#include "MainPage.xaml.h"
+#include <opencv2\imgproc\types_c.h>
+#include <opencv2\imgcodecs\imgcodecs.hpp>
+#include <opencv2\core\core.hpp>
+#include <opencv2\imgproc\imgproc.hpp>
+#include <opencv2\objdetect.hpp>
+#include <Robuffer.h>
+
+using namespace OpenCVExample;
+
+using namespace Platform;
+using namespace Windows::Foundation;
+using namespace Windows::Foundation::Collections;
+using namespace Windows::UI::Xaml;
+using namespace Windows::UI::Xaml::Controls;
+using namespace Windows::UI::Xaml::Controls::Primitives;
+using namespace Windows::UI::Xaml::Data;
+using namespace Windows::UI::Xaml::Input;
+using namespace Windows::UI::Xaml::Media;
+using namespace Windows::UI::Xaml::Navigation;
+
+using namespace Windows::UI::Xaml::Media::Imaging;
+using namespace Windows::Storage::Streams;
+using namespace Microsoft::WRL;
+
+
+// The Blank Page item template is documented at http://go.microsoft.com/fwlink/?LinkId=402352&clcid=0x409
+
+MainPage::MainPage()
+{
+	InitializeComponent();
+}
+
+void  OpenCVExample::MainPage::UpdateImage(const cv::Mat& image)
+{
+	// Create the WriteableBitmap
+	WriteableBitmap^ bitmap = ref new WriteableBitmap(image.cols, image.rows);
+
+	// Get access to the pixels
+	IBuffer^ buffer = bitmap->PixelBuffer;
+	unsigned char* dstPixels;
+
+	// Obtain IBufferByteAccess
+	ComPtr<IBufferByteAccess> pBufferByteAccess;
+	ComPtr<IInspectable> pBuffer((IInspectable*)buffer);
+	pBuffer.As(&pBufferByteAccess);
+
+	// Get pointer to pixel bytes
+	pBufferByteAccess->Buffer(&dstPixels);
+	memcpy(dstPixels, image.data, image.step.buf[1] * image.cols*image.rows);
+
+	// Set the bitmap to the Image element
+	img1->Source = bitmap;
+}
+
+// replace the image pane with an image in the "assets" folder, grpPC1.jpg
+void OpenCVExample::MainPage::btn1_Click(Platform::Object^ sender, Windows::UI::Xaml::RoutedEventArgs^ e)
+{
+	cv::Mat image = cv::imread("Assets/grpPC1.jpg");
+	Lena = cv::Mat(image.rows, image.cols, CV_8UC4);
+	cv::cvtColor(image, Lena, CV_BGR2BGRA);
+	UpdateImage(Lena);
+
+}
+
+// run a Canny filter on the image contained in "Lena", display the results to the image pane
+void OpenCVExample::MainPage::btn2_Click(Platform::Object^ sender, Windows::UI::Xaml::RoutedEventArgs^ e)
+{
+	//Canny filter
+	cv::Mat result;
+	cv::Mat intermediateMat;
+	cv::Canny(Lena, intermediateMat, 80, 90);
+	cv::cvtColor(intermediateMat, result, CV_GRAY2BGRA);
+	UpdateImage(result);
+}
+
+cv::String face_cascade_name = "Assets/haarcascade_frontalface_alt.xml";
+cv::CascadeClassifier face_cascade;
+
+cv::String body_cascade_name = "Assets/haarcascade_fullbody.xml";
+cv::CascadeClassifier body_cascade;
```

Comment:
These globals can be made locals to btn3_Click() and passed as const & to internalDetectObject().